MonstieWash/Assets/Scripts/Decoration/DecorationManager.cs

```diff
@@ -0,0 +1,523 @@
+    /// </summary>
+    private void CycleOptionsRight()
+    {
+        //Only allow cycling while enough decorations exist to matter and not holding an item.
+        if (m_barDecorations.Count <= 4) return;
+        if (managerStatus == ManagerStatus.Holding) return;
+        //Get last item and move to front
+        var itemToMove = m_barDecorations[m_barDecorations.Count-1];
+        m_barDecorations.RemoveAt(m_barDecorations.Count - 1);
+        m_barDecorations.Insert(0, itemToMove);
+        //Refresh Ui
+        RefreshBarUI();
+        //Instantly update new first items position.
+        m_barDecorations[0].sceneObject.transform.position = m_barDecorations[0].desiredLocation - new Vector3(decoBarBufferDist , 0);
+    }
+
+    /// <summary>
+    /// Called by releasing rotation input.
+    /// </summary>
+    private void ResetRotation()
+    {
+        m_rotatingValue = 0f;
+    }
+
+    /// <summary>
+    /// Takes a direction and updates scaling value via input.
+    /// </summary>
+    /// <param name="dir">-1 will rotate left, 1 will rotate right.</param>
+    private void ScaleInput(int dir)
+    {
+        if (m_currentlyHeldDecoration == null) return;
+        m_scaling = -dir;  
+    }
+
+    /// <summary>
+    /// Resets scaling value when input is cancelled.
+    /// </summary>
+    private void ResetScaling()
+    {
+        m_scaling = 0f;
+    }
+
+    /// <summary>
+    /// Correctly manages items position in status lists after being released.
+    /// </summary>
+    private void DropHeldItem()
+    {
+        if (m_currentlyHeldDecoration == null) return;      
+        m_activeDecorations.Remove(m_currentlyHeldDecoration);
+        m_barDecorations.Add(m_currentlyHeldDecoration);
+        m_currentlyHeldDecoration.returnToBar();
+        RefreshBarUI();
+        m_currentlyHeldDecoration.sceneObject.transform.position = m_currentlyHeldDecoration.desiredLocation;
+        m_currentlyHeldDecoration = null;
+        managerStatus = ManagerStatus.EmptyHand;
+    }
+
+    /// <summary>
+    /// Called whenever player inputs activate. Handles cases differently based on what player hand is closest to and if holding an object already.
+    /// </summary>
+    private void Clicked()
+    {
+        //Check you aren't already holding an item.
+        if (managerStatus == ManagerStatus.EmptyHand)
+        { 
+            //Find if any decorations in bar are near player hand first.
+            foreach (DecorationUi d in m_barDecorations)
+            {
+                if (Vector2.Distance(m_hand.transform.position, d.sceneObject.transform.position) <= m_pickupDistance)
+                {
+                    m_currentlyHeldDecoration = d;
+                    m_barDecorations.Remove(d);
+                    m_activeDecorations.Add(d);
+                    d.pickUp();
+                    RefreshBarUI();
+
+                    managerStatus = ManagerStatus.Holding;
+                    return;
+                }
```

Comment:
### 💡 **Suggestion**  | _This is a suggestion, feel free to take it or leave it._

This can defs be its own function if you want since it's duplicated. Just have it return a bool to exit early from this loop if successful.